packages/rum/test/readReplayPayload.ts

```diff
@@ -0,0 +1,40 @@
+import pako from 'pako'
+
+import type { Payload } from '@datadog/browser-core'
+import type { BrowserSegment } from '../src/types'
+import type { BrowserSegmentMetadataAndSegmentSizes } from '../src/domain/segmentCollection'
+
+export async function readReplayPayload(payload: Payload) {
+  return {
+    segment: await readSegmentFromReplayPayload(payload),
+    metadata: await readMetadataFromReplayPayload(payload),
+  }
+}
+
+function readSegmentFromReplayPayload(payload: Payload) {
+  return readJsonBlob((payload.data as FormData).get('segment') as Blob, {
+    decompress: true,
+  }) as Promise<BrowserSegment>
+}
+
+export async function readMetadataFromReplayPayload(payload: Payload) {
+  return readJsonBlob((payload.data as FormData).get('event') as Blob) as Promise<BrowserSegmentMetadataAndSegmentSizes>
+}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏since we don't use await in this function, we could remove the async keyword.
We could enforce that with [require-await](https://eslint.org/docs/latest/rules/require-await) eslint rule.